hasktorch/src/Torch/Optim.hs

```diff
@@ -37,6 +50,15 @@ gd' lr gradients depParameters dummy = (gd lr gradients depParameters, dummy)
 instance Optimizer GD where
     step = gd'
 
+-- TODO - alias sgd to gd once dependencies are migrated
+-- sgd = gd
+
+sgd :: Tensor -> [Parameter] -> [Tensor] -> [Tensor]
```

Comment:
```suggestion
sgd :: LearningRate -> [Parameter] -> [Tensor] -> [Tensor]
```

Reply:
Updated, although I'm hoping to deprecate sgd in this form (with Parameters as input) entirely.

Similar to the typed module - optimizer functions in Optim operate on tensors (instead of [Parameter]) and runStep bridges the marshalling between independent/dependent tensors. So for the examples, the typical pattern would be to call runStep with the selected optimizer instead of sgd directly.

I like this decoupling a little better because the optimizers can be used in other contexts outside backprop / computation graphs. As a side benefit this makes the examples less verbose and easier to grok for beginners since it hides the rituals around flattening / dependent tensors from the end user a little bit.